src/ast.rs

```diff
@@ -9598,9 +9599,9 @@ mod tests {
                                       volatile: false,
                                       const_: true,
                                   },
-                                  None),
+                                  Some(0)),
                     b"..."
-                }
+                }/*
```

Comment:
I believe that this test is supposed to be exercising the following parse:

```
<function-param> ::= fL <L-1 non-negative number> p <top-level CV-qualifiers> _
                 ::= fL 1 p <top-level CV-qualifiers> _                          # L = 2
                 ::= fL 1 p K _                                                  # CV-qualifiers = const
```

http://itanium-cxx-abi.github.io/cxx-abi/abi.html#mangle.function-param

I assume this test no longer passes since it is commented out? We should make sure it still works, and add a symbol to `tests.rs` containing this production.

Reply:
Alright I'll have to try to construct one of these symbols to see if they actually exist in real life.

Reply:
The example symbol given in the spec, `_Z1fIiEvT_PDtfL0pK_E`, does not parse in the binutils c++filt. I don't think this actually exists.

Reply:
On my MBP, the `abi::__cxa_demangle` implementation, which I think is the LLVM demangler, will handle it:

```
Demangling '_Z1fIiEvT_PDtfL0pK_E'
libiberty:    _Z1fIiEvT_PDtfL0pK_E
cxa:          void f<int>(int, decltype(fp)*)
demangle.go:  _Z1fIiEvT_PDtfL0pK_E
cpp_demangle: error: reference to a function arg that is either out-of-bounds, or in a context without function args
```

Given that we have been parsing it, I think we should continue.

Do you think it is worth raising an issue on the itanium ABI repo to discuss this?

Reply:
No, I don't.

After my changes we produce `void f<int>(int, decltype ({parm#1})*)` which has the right form at least. I'll restore the parsing tests.